### PR TITLE
ENH: Add array API standard v2022.12 support to numpy.array_api

### LINKED ISSUE #23880: ENH: Add 2022.12 support to numpy.array_api
### Proposed new feature or change:

This is a tracking issue for adding v2022.12 support to numpy.array_api. If possible this should happen before the 1.25 release (I hope I'm not too late for that), because that will make things much easier for people adopting the array API. 

The work should be straightforward. It mainly consists of making sure functions accept complex dtypes, although there are also a few new functions that need to be added (like `real` and `imag`).

### PR DESCRIPTION
**Pull request by asmeurer**
Fixes #23880

### CODE REVIEW

**Comment by asmeurer** [comment]
OK, not sure what happened with the draft status, but this is ready to review.

It would also be nice to include https://github.com/numpy/numpy/pull/23789 in the 1.25 release as well.

**Comment by asmeurer** [comment]
Actually, let me see how hard it is to add the `fft` extension here.

**Comment by asmeurer** [comment]
> Actually, let me see how hard it is to add the fft extension here.

The reason I'm hesitant is that we currently don't have any coverage of the fft extension in the array API test suite. As far as I know the standard fft API is identical to `np.fft`, but I'm not 100% sure about that.

I guess the biggest question is whether this is something that SciPy (or others?) will need soon for their own adding of array API support. @rgommers @tupui @tylerjereddy do you have any thoughts on how important it will be for SciPy's array API work for `numpy.array_api.fft` to be implemented (note that ffts already work the compat library because it's just `np.fft`)? Specifically whether it should be prioritized for NumPy 1.25 (to be released later this month). 

If it's not critical, I would suggest omitting it for now, and adding it later this year when we have full fft support in the test suite. But if it is indeed something that will be important for SciPy, I can add basic wrappers, with minimal testing.

File: numpy/array_api/_array_object.py [hunk 1]
```diff
@@ -454,11 +467,19 @@ def __bool__(self: Array, /) -> bool:
         # Note: This is an error here.
         if self._array.ndim != 0:
             raise TypeError("bool is only allowed on arrays with 0 dimensions")
-        if self.dtype not in _boolean_dtypes:
-            raise ValueError("bool is only allowed on boolean arrays")
         res = self._array.__bool__()
         return res
 
+    def __complex__(self: Array, /) -> float:
```
Comment (BvB93) [inline numpy/array_api/_array_object.py:473]:
```suggestion
    def __complex__(self: Array, /) -> complex:
```

**Comment by asmeurer** [comment]
Thank you Matti. If you or someone could merge https://github.com/numpy/numpy/pull/23789 too that would be great. I've already reviewed it and it's ready to go as far as I'm concerned.

**Pull request merged**